docs/source/blueprints/BP000.rst

```diff
@@ -0,0 +1,319 @@
+development processes. The idea is to have a lightweight document, very
+objective that will save us time in the medium and long run. We don’t have to
+overthink by trying to handle any possible scenario outside of ones we actually
+have a use case for. But we should also avoid the “auto-pilot” mode in our
+contributions.
+
+
+Specification
+*************
+
+One blueprint per topic
+-----------------------
+
+Try to follow the minimalist approach and be concise with content relevant to
+one particular topic. If you have a more general topic to discuss, you should
+set the type as “Epic Blueprint” (more below) but still try to be concise and
+focused on the subject.
+
+File format and location
+------------------------
+
+Our current documentation already uses ReStructuredText (.rst) format, so we
+will adopt .rst format here too. All blueprints will be located inside
+`docs/source/blueprints` with the filename `BPXXX.rst`, where XXX is the number
+of the blueprint. Just pick the next number available for your blueprint.
+
+Write for your audience
+-----------------------
+
+As mentioned before, your blueprint will be read by your peers, future members,
+and future yourself. Keep in mind that your audience is developers with a
+minimal understanding of the Avocado internals, but also be gentle and provide
+any context to understand the problem.
+
+Blueprints types
+----------------
+
+Currently, we have the following blueprint types:
+
+ * Architectural Blueprint: Any blueprint changing or introducing a new core
+   feature or architectural change to Avocado.
+
+ * Process Blueprint: Any blueprint that is not implementing a new core
+   feature, but changing how the project works. This could be, for instance,
+   related to the repositories or processes.
+
+ * Meta Blueprint: A blueprint about blueprints. Like this one and any future
+   blueprint that changes our blueprint's styles and methods.
+
+ * Epic Blueprint: A blueprint that is touching on multiple areas and is too
+   big to have all the documentation in one single blueprint. We could split
+   epic blueprints into smaller blueprints or issues (if they are small and
+   easy to understand).
+
+ * Component Blueprint: A blueprint with the intent to describe a new utility
+   module or a new plugin.
+
+Headers
+-------
+
+Python PEPs (Python Enhancement Proposals) uses RFC822 for describing the
+headers. This could be useful here too, especially when parsing those headers
+to display our list of blueprints with the current status.
+
+The current list of items of our blueprint headers is below:
+
+ * Number: Usually, the blueprint number in the format BPXXX
+
+ * Title: A short descriptive title, limited to 80 characters
+
+ * Author: The author or authors of blueprint
+
+ * Reviewers: All reviewers that approved and helped during the review process
+
+ * Created: Date string when the blueprint first draft was submitted. Please
+   use the following format: DD-MMM-YYYY.
```

Comment:
Why three M? Also, I would suggest that we use YYYY-MM-DD as an ISO standard that is more inclusive of American or other date formats that would start with the month and confuse themselves about this. I guess if it is not too late since we already have some blueprints out though.

Reply:
MMM is equivalent the first three characters of the month name since it is more readable for humans. This is the same format as PEPs.

Reply:
Alright, fair enough.